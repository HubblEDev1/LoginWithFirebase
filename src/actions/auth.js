import { createUserWithEmailAndPassword, getAuth, signInWithEmailAndPassword, signInWithPopup, signOut, updateProfile } from "@firebase/auth";
import Swal from "sweetalert2";
import { googleAuthProvider } from "../firebase/firebase-config";
import { types } from "../types/types"
import { finishLoading, startLoading } from "./ui";

/*export const login = (uid, displayName) => {
    return {
        type: types.login
    }
}*/
export const startLoginEmailPasswords = (email, password) => {
    return (dispatch) => {
        /*setTimeout(() => {
            dispatch(login(123, 'edwin'))
        }, 3500);*/
        dispatch(startLoading());
        const auth = getAuth();
        signInWithEmailAndPassword(auth, email, password).then( ({user}) => {
            console.log(user);
            dispatch(finishLoading());
            dispatch(login(user.uid, user.displayName));
        }).catch((e) => {
            console.log(e);
            dispatch(finishLoading());
            Swal.fire('Error', e.message, 'error');
        })
    }
}

export const startGoogleLogin = () => {
    return ( dispatch ) => {
        const auth = getAuth();
        signInWithPopup(auth, googleAuthProvider).then(({user}) => {
            dispatch(login(user.uid, user.displayName));
        });
    }
}

export const startRegisterWithEmailPasswordName = (email, pass, name) => {
    return (dispatch) => {
        const auth = getAuth();
        createUserWithEmailAndPassword(auth, email, pass).then( async ({user}) => {
            await updateProfile(auth.currentUser, {displayName: name});
            dispatch(
                login(user.uid, user.displayName)
            );
        }).catch( e => {
            console.log(e);
            Swal.fire('Error', e.message, 'error');
        })
    }
}


export const login = (uid, displayName) => ({
    type: types.login,
    payload: {
        uid, 
        displayName
    }
}); 

export const startLogout = () => {
    return async ( dispatch ) => {
        const auth = getAuth();
        signOut(auth);
        dispatch( logout() );
    }
}

export const logout = () => ({
    type: types.logout,
})