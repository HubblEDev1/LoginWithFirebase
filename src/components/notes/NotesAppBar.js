import React from 'react'

export const NotesAppBar = () => {
    return (
        <div className="notes__appbar">
            <span>20 August 1998</span>
            <div>
                <button className="btn">
                    Picture
                </button>
                <button className="btn">
                    Save
                </button>
            </div>
        </div>
    )
}
