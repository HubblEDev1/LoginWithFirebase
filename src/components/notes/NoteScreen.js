import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    return (
        <div
            className="notes__main-content"
        >
            <NotesAppBar />
            <div className="notes__content">
                <input
                    type="text"
                    placeholder="Some awesome title"
                    className="notes__title-input"
                />

                <textarea
                    placeholder="What happend today"
                    className="notes__textarea"
                >
                </textarea>

                <div className="notes__image">
                    <img
                        src="https://i.pinimg.com/originals/d8/cb/3c/d8cb3ce047d28819c65c4c5105fa30b3.jpg"
                        alt="Image"
                    />
                </div>

            </div>
        </div>
    )
}
