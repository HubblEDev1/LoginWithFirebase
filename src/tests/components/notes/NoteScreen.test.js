import configureStore from 'redux-mock-store'; //ES6 modules
import thunk from 'redux-thunk';
import '../../../setupTests';

import { mount } from "enzyme";
import { Provider } from "react-redux";

import {NoteScreen} from '../../../components/notes/NoteScreen';
import { activeNote, startNewNote } from '../../../actions/notes';

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn(),
}));

jest.mock('../../../actions/auth', () => ({
    startLogout: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '1', 
        name: 'Ed'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 1234,
            title: 'Hola',
            body: 'mundo', 
            date: 0
        },
        notes: []
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store = { store }>
        <NoteScreen/>
    </Provider>
);

describe('Pruebas en NoteScreen', () => {
    test('should ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should dispatch active note', () => {
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hi again'
            }
        });

        expect( activeNote ).toHaveBeenCalledWith(
            1234,
            {
                body: 'mund',
                title: 'Hi again',
                id: 1234,
                date: 0
            }
        );
        /*
        expect( activeNote ).toHaveBeenCalledWith(
            1234,
            {
                body: 'mundo',
                title: 'Hi again',
                id: 1234,
                date: 0
            }
        );*/
    })
    
})
