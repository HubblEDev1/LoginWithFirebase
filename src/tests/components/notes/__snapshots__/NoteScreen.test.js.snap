// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en NoteScreen should  1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      20 August 1998
    </span>
    <input
      id="fileSelector"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div
      style={
        Object {
          "marginRight": "30px",
        }
      }
    >
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Hola"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happend today"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
